src/query/interpreter.cpp

```diff
@@ -1004,15 +1004,12 @@ Callback HandleReplicationQuery(ReplicationQuery *repl_query, const Parameters &
   evaluation_context.timestamp = QueryTimestamp();
   evaluation_context.parameters = parameters;
   auto evaluator = PrimitiveLiteralExpressionEvaluator{evaluation_context};
-
+  auto const is_coordinator_managed = flags::CoordinationSetupInstance().IsCoordinatorManaged();
   Callback callback;
   switch (repl_query->action_) {
     case ReplicationQuery::Action::SET_REPLICATION_ROLE: {
 #ifdef MG_ENTERPRISE
-      if (FLAGS_coordinator_id) {
-        throw QueryRuntimeException("Coordinator can't set roles!");
-      }
-      if (FLAGS_management_port) {
+      if (is_coordinator_managed) {
```

Comment:
Why are we using 2 checks:
1. is_coordinator_managed
2. IsCoordinator and IsDataInstances

Why not rely just on one, for example check IsCoordinator

Reply:
fix by sending coordinator state